### ISSUE TITLE
fix renovate configuration behavior and failures

### ISSUE THREAD

**Issue by adrielp** [liatrio/liatrio-otel-collector#244]
# Overview 

The Renovate configuration is currently failing. The workflows are causing issues. Renovate should have it's dependencies grouped such that the workflows are fixed and minimal PR's are open. 

Additionally, each fix commit would create a new version & thus we want to group these dependencies fixes together as much as possible.

**Comment by adrielp** [comment]
* Deleted the dependabot configuration that was there 
* updated labeler as well while I was there https://github.com/liatrio/liatrio-otel-collector/pull/258
* potentially going to create an ADR with a couple things:
    * update configuration for otel repo to enable renovate to bypass main branch detection for auto merges
* working on figuring out how to best update the match managers to ensure compatibility with the `make tidy-all` functionality that is run as part of the workflows.

**Comment by adrielp** [comment]
* updated latest versions of all components
* removed `automerge` from the `renovate.json`
* adjusted the `renovate.json` to use file match managers 
    * segmented out into compgen, components, tools, actions, and dockerfiles
* still noticed that one pr that was open still didn't succeed. Looks like renovate didn't update the packages that needed to be updated alongside through `go mod tidy` - going to review this with the team

**Comment by adrielp** [comment]
This [pull request](https://github.com/liatrio/liatrio-otel-collector/pull/268) is an example of a cyclical dependency. Every time we publish a new version of the collector, Renovate is going to create a PR to update this version, which in turn will cause a new version to be created, which will cause a new PR and so forth. 

OTEL manages this in two ways. 
1) OTEL has a separate release process that isn't on every commit and is managed by a separate repository. 
2) OTEL leverages the [multimod](https://github.com/open-telemetry/opentelemetry-go-build-tools/tree/main/multimod) utility to manage versions via the `versions.yaml` file and publish commits. 

Currently my thought process is to simply exclude our own package, ensuring that the replace statement always keeps it up-to-date and not creating a cyclical dependency.

**Comment by adrielp** [comment]
Giving this a couple more days to see if any packages are updated. Will be waiting for v0.93.0 comes out for collector contrib.

**Comment by adrielp** [comment]
Renovate usage has improved some since the last set of changes, but is still subject to failing due to not having run `go tidy` or `make tidy-all`. 

I tested out adding the [postUpdateOptions](https://github.com/liatrio/liatrio-otel-collector/pull/268#issuecomment-1908806934) for renovate but it caused an issue because we're using `v1.21` of go instead of the currently supported `v1.20`.  This is reflective by [this renovate comment](https://github.com/liatrio/liatrio-otel-collector/pull/268#issuecomment-1908806934) on one of the PR's it opened. 

For now we've reverted this change and will be manually fixing PR's as needed. We will create new tickets per the last TAG sync that evaluates whether or not we want to self-host renovate for this repository and leverage the [postUpgradeTasks](https://docs.renovatebot.com/configuration-options/#postupgradetasks) instead.
